OPCUA/OPCUAGetTaskHandler.cs

```
@@ -0,0 +1,53 @@
+﻿namespace BoxwriterResmarkInterop.OPCUA;
+
+using System.Text.RegularExpressions;
+
+using Interfaces;
+
+using Mediator;
+
+using TCP;
+
+public class OPCUAGetTaskHandler : IRequestHandler<GetTaskRequest, StringResponse>
+{
+    private readonly ILogger<OPCUAGetTaskHandler> _logger;
+    private readonly IOPCUAService _opcuaService;
+
+    public OPCUAGetTaskHandler(ILogger<OPCUAGetTaskHandler> logger, IOPCUAService opcuaService)
+    {
+        _logger = logger;
+        _opcuaService = opcuaService;
+    }
+
+    public async ValueTask<StringResponse> Handle(GetTaskRequest request, CancellationToken cancellationToken)
+    {
+        //Extract printer id
+        var data = request.data;
+
+        var printerIdRegex = new Regex(@",\s*(.+)\s*}").Match(data);
+
+        if (!printerIdRegex.Success)
+        {
+            _logger.LogError("Unable to find printer ID. Request was {data}", data);
+
+            throw new InvalidDataException($"Unable to find printer ID. Request was {data}.");
+        }
+
+        var printerId = printerIdRegex.Groups[1].Value;
+
+        //Get stored message list from the printer
+        var response = await _opcuaService.CallMethodAsync(printerId, "GetStoredMessageList", cancellationToken);
+
+        //Return formatted response
+        var outputArguments = response?[0]?.OutputArguments;

```

I feel like there is a lot of excess nullability here. Maybe just me. What do you do if any of it is null?

---

WTF does `response?[0]?` even mean?  I think a better approach is:

```
if (response is null)
{
    // Do some null handling
}

var outputArguments = response.First()?.OutputArguments ?? throw new IndexOutOfRangeException();
```

---

`response.First()?` should be `OrDefault` otherwise it will throw.

---

Or, if you just want to throw, just let any of this throw on its own instead of doing all this null checking.